src/edfi-paging-test/tests/helpers/test_request_logger.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+
+from pandas import DataFrame
+import pytest
+
+from edfi_paging_test.helpers.measurement import Measurement
+from edfi_paging_test.helpers.request_logger import get_DataFrame, log_request
+
+
+MEASUREMENT_1 = Measurement(
+    "academicWeeks",
+    "https://localhost/WebApi/data/v3/ed-fi/academicWeeks",
+    1,
+    200,
+    30,
+    40,
+    200,
+)
+MEASUREMENT_2 = Measurement(
+    "assignments",
+    "https://localhost/WebApi/data/v3/lmsx/assignments",
+    3,
+    500,
+    500,
+    321,
+    200,
+)
+
+
+def describe_when_logging_a_request() -> None:
+    def describe_given_there_is_already_an_item() -> None:
+        @pytest.fixture()
+        def logs(mocker) -> List[Measurement]:
+            request_log = [MEASUREMENT_1]
+            mocker.patch(
+                "edfi_paging_test.helpers.request_logger._request_log", request_log
```

Comment:
@vimayya if you ever need to mock a module-level variable or function (outside of a class/object), then this `mocker` object is very useful, from the library `pytest-mock`. You can create a `patch` without it, but you'll also need to restore the original. With the `mocker` object, the original version of the variable or function is automatically restored.